docs/auth/vmware-cloud/provider.md

```diff
@@ -0,0 +1,250 @@
+   - `Define Scopes`: check `OpenID` at the bottom.
+1. Click **Create**.
+1. Take note of the `App ID` in the resulting modal; this is the client ID to be
+   used by Backstage.
+
+## Install the provider in the backend
+
+### New backend system
+
+Apps using the [new backend system](../../backend-system/index.md),
+can enable the VMware Cloud provider with a small modification like:
+
+```ts title="packages/backend-next/src/index.ts"
+import { createBackend } from '@backstage/backend-defaults';
+
+const backend = createBackend();
+backend.add(import('@backstage/plugin-auth-backend'));
+/* highlight-add-start */
+backend.add(
+  import('@backstage/plugin-auth-backend-module-vmware-cloud-provider'),
+);
+/* highlight-add-end */
+backend.start();
+```
+
+### Old backend system
+
+This provider was added after the migration of the auth-backend plugin to the
+new backend system, so no default provider factory was added. Because of this,
+the installation procedure for old-style backends is slightly more involved:
+
+```ts title="packages/backend/src/plugins/auth.ts"
+import {
+  DEFAULT_NAMESPACE,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import {
+  createRouter,
+  providers,
+  defaultAuthProviderFactories,
+} from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+/* highlight-add-start */
+import {
+  commonSignInResolvers,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import {
+  vmwareCloudAuthenticator,
+} from '@backstage/plugin-auth-backend-module-vmware-cloud-provider';
+/* highlight-add-end */
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  return await createRouter({
+    logger: env.logger,
+    config: env.config,
+    database: env.database,
+    discovery: env.discovery,
+    tokenManager: env.tokenManager,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+      /* highlight-add-start */
+      vmwareCloudServices: createOAuthProviderFactory({
+        authenticator: vmwareCloudAuthenticator,
+        signInResolver:
+          commonSignInResolvers.emailLocalPartMatchingUserEntityName(),
+      }),
+      /* highlight-add-end */
+```
+
+In the above, `commonSignInResolvers.emailLocalPartMatchingUserEntityName()`
+can be replaced with a more suitable resolver for the app in question.
+
+## Configure Sign-in Resolution
+
+See [Sign-in Identities and Resolvers](../identity-resolver.md) for details.
+
+## Create a Utility API
+
+At the moment, this auth provider is not included in the default API factories,
+but you can create one with a change to your frontend like:
+
+```ts title="packages/app/src/apis.ts"
+import {
+  ScmIntegrationsApi,
+  scmIntegrationsApiRef,
+  ScmAuth,
+} from '@backstage/integration-react';
+import {
+  costInsightsApiRef,
+  ExampleCostInsightsClient,
+} from '@backstage/plugin-cost-insights';
```

Comment:
Fair amount of this example isn't related to what's being done, best trim this down a bit

Reply:
Removed in 9185966